enhancements/installer/operators-check.md

```diff
@@ -0,0 +1,174 @@
+* Installer exits non-zero in those situations and delivers meaningful error message
+
+### Non-Goals
+
+* Installer handling other classes of errors, such as failure to provision machines by MAO
+
+## Proposal
+
+### Workflow Description
+
+Cluster admin will begin installing cluster as usual. The Installation workflow will be:
+
+1. Cluster initializes as usual
+2. As usual, installer checks that cluster version is Available=true Progressing=False Degraded=False 
+3. Installer checks status of each cluster operator for stability
+4. If a cluster operator is not stable the installer logs a message and throws an error
+
+### API Extensions
+
+None
+
+### Implementation Details/Notes/Constraints [optional]
+
+The important detail is how we define _stable_. The definition proposed here is:
+
+```
+If a cluster operator does not maintain Progressing=False for at least 30 seconds,
+during a five minute period it is unstable.
+```
+
+### Risks and Mitigations
+
+One risk would be a false positive: the Installer identifies that a cluster
+operator is unstable, but it turns out the operator is perfectly healthy;
+the install was declared a failure but was actually successful. This risk
+seems low and a risk that could be managed by monitoring these specific failures
+in CI.
+
+### Drawbacks
+
+This enhancement has some significant potential drawbacks in terms of design:
+
+A fundamental design principle of 4.x and the operator framework has been
+delegating responsibility away from the Installer to allow clean separation
+of concerns and better maintainbility. Without this enhancement, it is the
+responsibility of the Cluster-Version Operator to determine whether given
+Cluster Operators constitute a successful version. The idea of keeping the
+cluster-version operator as the single responsible party is discussed in the
+alternatives section.
+
+Following the previous point, this enhancement could introduce a slippery slope.
+Where before this enhancement, the Installer would simply query the CVO; now we
+are adding additional conditional logic. Introducing this conditional logic leads
+to the risk of further conditions, which could make the install process more brittle
+and introduce the potential for false positives or other failures.
+
+Does implementing this enhancement address symptoms of issues with operator status definitions?
+Should operators themselves be setting Degraded=True when they don't meet this stability criterion?
+
+As we have seen with other timeouts in the Installer, developers and users will want to change these.
+We should define a process for refining our stability criteria. 
+
+## Design Details
+
+### Open Questions [optional]
+
+1. What is the correct definition of a stable operator? More importantly, how can we
+refine this definition?
+2. Should this logic belong in the Installer, CVO, or another component?
+
+### Test Plan
+
+This code would go directly into the installer and be exercised by e2e tests.
```

Comment:
By a synthetic CO that immediately becomes Available=True fulfilling the historic requirement for install-complete but maintains Progressing=True for 30 seconds after? Do we have tests that watch the cluster during the install or is this new ground? I think @deads2k has called these observers?

Reply:
> I think @deads2k has called these observers?

We don't have e2e observers yet, but if they existed it could work.  I built a backstop into our pre-test CI step that we can change from "wait five minutes to settle" to "fail if anything is not settled.  If it trips, this feature has a bug.